### PR TITLE
#38 - Add the ability to change theme in profile

### PR DESCRIPTION
**Pull request by ImmortalIdiot**
In this pr I add the ability to change theme in profile. To do this, I had to define a theme in ProfileScreen and provide it (theme) into all screens from MainActivity. Also, I had to replace the color palette with the MaterialTheme colors IN ALL FILES (Oh, God, I really could do it and not start smoking)

### CODE REVIEW

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/LoginScreen.kt [hunk 1]
```diff
@@ -127,37 +125,35 @@ fun LoginScreen(
...
-                    color = palette.outline,
+                    color = scheme.outline,
                     shape = roundedShape
                 )
                 .padding(vertical = dimensions.verticalBigPadding)
         ) {
             Column(
-                modifier = modifier.fillMaxWidth(0.85f),
+                modifier = modifier
+                    .fillMaxWidth(0.85f),
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/LoginScreen.kt:138]:
```suggestion
                modifier = modifier.fillMaxWidth(0.85f),
```

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 2]
```diff
@@ -167,17 +167,13 @@ fun SignUpScreen(
                 horizontalAlignment = Alignment.CenterHorizontally,
                 verticalArrangement = Arrangement.Center
             ) {
-                BoxLabel(
-                    text = "Регистрация",
-                    palette = palette
-                )
+                BoxLabel(text = "Регистрация")
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:170]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 3]
```diff
@@ -167,17 +167,13 @@ fun SignUpScreen(
...
                 Spacer(modifier = modifier.height(dimensions.verticalXXLarge))
                 if (state is SignUpViewModel.State.Init ||
                     state is SignUpViewModel.State.SignUpValidationFirstPartError
                 ) {
                     StudentIdTextField(
                         hint = "Номер студенческого билета",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:176]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 4]
```diff
@@ -196,18 +192,18 @@ fun SignUpScreen(
                             Text(
                                 text = "Email",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:193]: again

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 5]
```diff
@@ -217,10 +213,9 @@ fun SignUpScreen(
                     Spacer(modifier = modifier.height(dimensions.verticalXLarge))
                     PasswordField(
                         hint = "Пароль",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:215]: again

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 6]
```diff
@@ -248,7 +243,7 @@ fun SignUpScreen(
                         modifier = modifier
                             .fillMaxHeight(0.18f)
                             .fillMaxWidth(0.55f),
-                        palette = palette,
+                        scheme = scheme,
                         text = "Далее",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:247]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 7]
```diff
@@ -271,18 +266,18 @@ fun SignUpScreen(
                             Text(
                                 text = "ФИО",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:267]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 8]
```diff
@@ -302,18 +297,18 @@ fun SignUpScreen(
                             Text(
                                 text = "Группа",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:298]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 9]
```diff
@@ -323,7 +318,7 @@ fun SignUpScreen(
                     Spacer(modifier = modifier.height(dimensions.verticalXLarge))
                     PrimaryButton(
                         modifier = modifier.fillMaxHeight(0.13f),
-                        palette = palette,
+                        scheme = scheme,
                         text = "Зарегистрироваться",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:322]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 10]
```diff
@@ -341,12 +336,11 @@ fun SignUpScreen(
                 ) {
                     Text(
                         text = "Есть аккаунт?",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:338]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt [hunk 11]
```diff
@@ -341,12 +336,11 @@ fun SignUpScreen(
...
-                        style = boldInter14.copy(color = palette.text),
+                        style = boldInter14.copy(color = scheme.onPrimary),
                     )
                     RedirectText(
                         modifier = modifier,
                         text = "Войдите",
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/auth/SignUpScreen.kt:343]: string resources

File: app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt [hunk 12]
```diff
@@ -1,37 +1,256 @@
...
+                //TODO(): open the change group dialog
+            }
+        )
+        Spacer(modifier = modifier.height(dimensions.verticalStandard))
+        PrimaryButton(
+            modifier = modifier.fillMaxWidth(0.8f),
+            borderWidth = dimensions.borderXSSmall,
+            shape = RoundedCornerShape(dimensions.shapeMLarge),
+            scheme = scheme,
+            text = stringResource(id = R.string.delete_account),
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/presentation/screens/main/ProfileScreen.kt:236]: А здесь string resource ты уже используешь.... Что ж не везде то?

File: app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ThemeViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+object ThemeManager {
+    val themeFlow: MutableStateFlow<ThemeStyle> = MutableStateFlow(ThemeStyle.AUTO)
+}
+
+enum class ThemeStyle {
+    DARK,
+    LIGHT,
+    AUTO
+}
```
[the remark this answers is not in the document]
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/presentation/viemodels/main/ThemeViewModel.kt:44]: Зачем запятая после AUTO и точка с запятой?

**Pull request merged**